cmd/minishift/cmd/start_preflight.go

```diff
@@ -447,6 +447,10 @@ func checkIsoUrl() bool {
 		return false
 	}
 
+	if strings.HasPrefix(isoUrl, "http://") || strings.HasPrefix(isoUrl, "https://") {
```

Comment:
So what's about adding some unit tests here? This is a pure string function which can be fully tested. 

Personally, I'd love to see the URL string being passed to _checkIsoUrl_, so that you can easily write tests. If that is not possible, you can also call _viper.SetString_ as part of your test setup and keep the code as is. Either way, it should be easy to add a classic Go table driven test - https://github.com/golang/go/wiki/TableDrivenTests - for this exercising the various cases. 

The test also helps to document the expected values for these URLs, aka shortnames, file path and URLs.

Reply:
well not entirely, we do check if the file exists on disk if its a file path. https://github.com/minishift/minishift/blob/c677a9853b13c96e6353bacd809da5648d20375f/cmd/minishift/cmd/start_preflight.go#L493-L497